lib/netext/transport.go

```diff
@@ -0,0 +1,229 @@
+package netext
+
+import (
+	"bytes"
+	"encoding/base64"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/ThomsonReutersEikon/go-ntlm/ntlm"
+	"github.com/loadimpact/k6/lib"
+	"github.com/loadimpact/k6/stats"
+	"github.com/pkg/errors"
+)
+
+type Transport struct {
+	*http.Transport
+	options   lib.Options
+	tags      map[string]string
+	trail     *Trail
+	tlsInfo   TLSInfo
+	samplesCh chan<- stats.SampleContainer
+
+	// ntlm authentication
+	mu              sync.Mutex
+	ntlmAuthCache   map[string]bool
+	enableNtlmCache bool
+}
+
+func NewTransport(transport *http.Transport, samplesCh chan<- stats.SampleContainer, options lib.Options, tags map[string]string) *Transport {
+	return &Transport{
+		Transport:       transport,
+		tags:            tags,
+		options:         options,
+		samplesCh:       samplesCh,
+		ntlmAuthCache:   make(map[string]bool),
+		enableNtlmCache: true,
+	}
+}
+
+func (t *Transport) CloseIdleConnections() {
+	t.enableNtlmCache = false
```

Comment:
This seems like the wrong semantic here. `CloseIdleConnections()` should probably just clear the cache, not disable its future use. The `enableNtlmCache` should probably be controlled by something like the `Transport.DisableKeepAlives`, if at all.

Reply:
Agree, it feels like this should clear rather than disable the cache.